From 0811cef8829d6e0fe28e4f1e73685a51bddd97a5 Mon Sep 17 00:00:00 2001
From: "Singh, Sapna1" <sapna1.singh@intel.com>
Date: Thu, 10 Nov 2022 15:20:12 +0530
Subject: [PATCH] [W/A] Fix for Camera Asking Location Permission every time

Camera app prompts for "Tag your photos and videos with the locations
where they're taken" permission on every open.

If the lastversion value is less than 2, it will force the location
choice to user on every camera open.

Set the lastversion value to 2 to skip forcing the location choice
to user.

Change-Id: Icff3ca5db8a2c34732885694d620f51dd22d2931
Tracked-On: OAM-104683
Signed-off-by: Singh, Sapna1 <sapna1.singh@intel.com>
---
 src/com/android/camera/settings/AppUpgrader.java      | 4 +++-
 src/com/android/camera/settings/SettingsUpgrader.java | 3 +++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/com/android/camera/settings/AppUpgrader.java b/src/com/android/camera/settings/AppUpgrader.java
index cce750ce5..c499511f5 100644
--- a/src/com/android/camera/settings/AppUpgrader.java
+++ b/src/com/android/camera/settings/AppUpgrader.java
@@ -122,9 +122,11 @@ public class AppUpgrader extends SettingsUpgrader {
     public void upgrade(SettingsManager settingsManager, int lastVersion, int currentVersion) {
         Context context = mAppController.getAndroidContext();
         CameraPictureSizesCacher.QueriedSizeForFirstTime = false;
+        // Setting lastversion to 2 to avoid getting location tag permission on
+        // every camera open.
+        lastVersion = 2;
         // Do strings upgrade first before 'earlier' upgrades, since they assume
         // valid storage of values.
-        lastVersion = 0;
         if (lastVersion < CAMERA_SETTINGS_STRINGS_UPGRADE) {
             upgradeTypesToStrings(settingsManager);
         }
diff --git a/src/com/android/camera/settings/SettingsUpgrader.java b/src/com/android/camera/settings/SettingsUpgrader.java
index b6667d79f..00ac535df 100644
--- a/src/com/android/camera/settings/SettingsUpgrader.java
+++ b/src/com/android/camera/settings/SettingsUpgrader.java
@@ -50,6 +50,9 @@ public abstract class SettingsUpgrader {
      */
     public void upgrade(SettingsManager settingsManager) {
         int lastVersion = getLastVersion(settingsManager);
+        // It is required to call upgrade function everytime to support dynamic client
+        // update so that on every camera open,supported parameters should be read
+        // instaed of using the cached camera parameters.
         //if (lastVersion != mTargetVersion) {
             upgrade(settingsManager, lastVersion, mTargetVersion);
         //}
-- 
2.37.0

